### PR TITLE
Limit variable scope

### PR DESCRIPTION
**Pull request by practicalswift**
Limit variable scope as much as possible in order to:
* improve code readability,
* reduce complexity, and/or
* avoid unnecessary initialisations.

### CODE REVIEW

**Comment by sipa** [comment]
utACK 5e6c94531046135d406ea755851fd614a76fb543

File: src/policy/fees.cpp [hunk 1]
```diff
@@ -855,7 +855,7 @@ bool CBlockPolicyEstimator::Read(CAutoFile& filein)
     try {
         LOCK(cs_feeEstimator);
         int nVersionRequired, nVersionThatWrote;
-        unsigned int nFileBestSeenHeight, nFileHistoricalFirst, nFileHistoricalBest;
+        unsigned int nFileBestSeenHeight;
```
Comment (benma) [inline src/policy/fees.cpp]: move down to the before L865, to move it closer to its first use.

File: src/qt/trafficgraphwidget.cpp [hunk 2]
```diff
@@ -48,12 +48,12 @@ int TrafficGraphWidget::getGraphRangeMins() const
 void TrafficGraphWidget::paintPath(QPainterPath &path, QQueue<float> &samples)
 {
     int h = height() - YMARGIN * 2, w = width() - XMARGIN * 2;
```
Comment (benma) [inline src/qt/trafficgraphwidget.cpp]: `h`, `w`, `x` could be moved into the if clause

File: src/rpc/mining.cpp [hunk 3]
```diff
@@ -98,13 +98,12 @@ UniValue getnetworkhashps(const JSONRPCRequest& request)
...
 {
     static const int nInnerLoopCount = 0x10000;
-    int nHeightStart = 0;
     int nHeightEnd = 0;
     int nHeight = 0;
 
     {   // Don't keep cs_main locked
         LOCK(cs_main);
-        nHeightStart = chainActive.Height();
+        int nHeightStart = chainActive.Height();
```
Comment (benma) [inline src/rpc/mining.cpp]: is this var even needed? could do `nHeight = chainActive.Height();` and use `nHeight` instead of `nHeightStart` on the line after.

File: src/script/interpreter.cpp [hunk 4]
```diff
@@ -263,11 +263,11 @@ bool EvalScript(std::vector<std::vector<unsigned char> >& stack, const CScript&
     set_error(serror, SCRIPT_ERR_UNKNOWN_ERROR);
     if (script.size() > MAX_SCRIPT_SIZE)
         return set_error(serror, SCRIPT_ERR_SCRIPT_SIZE);
-    int nOpCount = 0;
     bool fRequireMinimal = (flags & SCRIPT_VERIFY_MINIMALDATA) != 0;
```
Comment (benma) [inline src/script/interpreter.cpp]: why did you move only `nOpCount` but not this or some of the other vars above?

**Review by benma — CHANGES_REQUESTED** [review]

File: src/script/interpreter.cpp (hunk 4 above)
Comment (practicalswift) [inline src/script/interpreter.cpp]: Yes, there are more vars that could be moved into the tighter `try` scope. I'll take all `interpreter.cpp` changes in a separate PR since they'll need a more thorough review. I've now removed the `nOpCount` move from this PR.

**Comment by practicalswift** [comment]
@benma Feedback items addressed! Thanks for the review.

File: src/qt/trafficgraphwidget.cpp [hunk 5]
```diff
@@ -47,13 +47,14 @@ int TrafficGraphWidget::getGraphRangeMins() const
 
 void TrafficGraphWidget::paintPath(QPainterPath &path, QQueue<float> &samples)
 {
-    int h = height() - YMARGIN * 2, w = width() - XMARGIN * 2;
-    int sampleCount = samples.size(), x = XMARGIN + w, y;
+    int sampleCount = samples.size();
     if(sampleCount > 0) {
+        int w = width() - XMARGIN * 2;
+        int x = XMARGIN + w, h = height() - YMARGIN * 2;
```
Comment (benma) [inline src/qt/trafficgraphwidget.cpp]: a super-nit, but I preferred `w` and `h` on the same line as before, because of the symmetry.

**Comment by benma** [comment]
utACK 4adc8389af964e84c1f97bc28707904341bbbc74

**Comment by practicalswift** [comment]
@benma Good point. `w` and `h` now on the same line again :-)

File: src/qt/trafficgraphwidget.cpp [hunk 6]
```diff
@@ -47,13 +47,14 @@ int TrafficGraphWidget::getGraphRangeMins() const
 
 void TrafficGraphWidget::paintPath(QPainterPath &path, QQueue<float> &samples)
 {
-    int h = height() - YMARGIN * 2, w = width() - XMARGIN * 2;
-    int sampleCount = samples.size(), x = XMARGIN + w, y;
+    int sampleCount = samples.size();
     if(sampleCount > 0) {
+        int h = height() - YMARGIN * 2, w = width() - XMARGIN * 2;
+        int x = XMARGIN + w;
```
Comment (benma) [inline src/qt/trafficgraphwidget.cpp]: Sorry for another nit, but why not inline this `x` (`YMARGIN + h` is also inlined`)? Then you can declare `int x` on line 56 (or inline those two as well).

Currently, there something like shadowing going on, where `x` is not modified inside the loop, as they are independent.

**Comment by jtimon** [comment]
utACK 90593ed92cfd5d49900a6fb6c2c10a482ab9fdbb

**Comment by paveljanik** [comment]
utACK 90593ed

File: src/qt/trafficgraphwidget.cpp (hunk 6 above)
Comment (practicalswift) [inline src/qt/trafficgraphwidget.cpp]: @benma I'm not sure I follow. The value of `x` assigned in the looped is used outside of the loop, so declaring it on line 56 wouldn't maintain the same logic as before, would it? Please post a gist with your suggested formulation :-)

Comment (benma) [inline src/qt/trafficgraphwidget.cpp]: You are correct, sir! Apologies :)

Comment (practicalswift) [inline src/qt/trafficgraphwidget.cpp]: @benma Apology accepted! 😄
